### DISCUSSION TITLE
Enhancing the navigation widgets by home-grown code

### DISCUSSION THREAD

**Discussion by j8xixo12** [solvcon/modmesh#340] [Ideas]
The Euler1d app has been integrated with the matplotlib navigation toolbar. However, this navigation widget is a third-party dependency from matplotlib, making it difficult to customize.

We should developing a custom navigation widget from scratch to allow for full UI customization.

**Comment by j8xixo12** [comment]
Navigation widget is a key feature of data visualization application, to make a home-grown navigation widget indeed requires a lot of effort, here is the development plan of nnhancing the navigation widgets by home-grown code, it will be separated into short-term and long-term part:

The short-term navigation widgets development plan:
- Add a indicator that allow mouse pointing to show mesh and solution information using native Qt (C++) on the top of current matplotlib code.

The long-term navigation widgets development plan is using native Qt (C++) to implement whole plotting system to instead of matplotlib, the development plan is:
- Implement Line and text object class that allow user to customize it (color, width and position)
- Base on home-grown Line and text object to create a plot class that draws xy-plot.

**Reply by yungyuc** [reply]
I turn this issue into a discussion conversation since we have not made any progress.  I think we should use more discussions.
